sda-commons-server-auth/src/test/java/org/sdase/commons/server/auth/key/RsaKeyLoaderSchedulerTest.java

```diff
@@ -0,0 +1,103 @@
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class RsaKeyLoaderSchedulerTest {
+
+  private final RsaPublicKeyLoader keyLoader = new RsaPublicKeyLoader();
+  private final ScheduledExecutorService executorService =
+      Executors.newSingleThreadScheduledExecutor();
+  private final RsaKeyLoaderScheduler keyLoaderScheduler =
+      RsaKeyLoaderScheduler.create(keyLoader, executorService);
+  private boolean elseCondition = false;
+
+  @Test
+  public void shouldNotBeAuthenticatedWithRevokedKey() {
+    // given
+    final RSAPublicKey validKey = Mockito.mock(RSAPublicKey.class);
+    final String kidForValidKey = "kidForValidKey";
+    final RSAPublicKey keyToBeRevoked = Mockito.mock(RSAPublicKey.class);
+    final String kidForKeyToBeRevoked = "kidForKeyToBeRevoked";
+
+    KeySource keySource =
+        new KeySource() {
+          @Override
+          public List<LoadedPublicKey> loadKeysFromSource() throws KeyLoadFailedException {
+            if (!elseCondition) {
+              return Arrays.asList(
+                  new LoadedPublicKey(kidForValidKey, validKey, this),
+                  new LoadedPublicKey(kidForKeyToBeRevoked, keyToBeRevoked, this));
+            } else {
+              return singletonList(new LoadedPublicKey("kidForValidKey", validKey, this));
+            }
+          }
+        };
+
+    keyLoader.addKeySource(keySource);
+    keyLoaderScheduler.start(0, 2, TimeUnit.SECONDS);
+    await()
+        .untilAsserted(
+            () -> {
+              assertThat(keyLoader.getKey(kidForValidKey)).isSameAs(validKey);
+              assertThat(keyLoader.getKey(kidForKeyToBeRevoked)).isSameAs(keyToBeRevoked);
+            });
+
+    // when -> simulate revoking of key:
+    elseCondition = true;
+
+    // then
+    await()
+        .untilAsserted(
+            () -> {
+              assertThat(keyLoader.getKey(kidForValidKey)).isSameAs(validKey);
+              // revoked key not found anymore
+              assertThat(keyLoader.getKey(kidForKeyToBeRevoked)).isNull();
+            });
+  }
+
+  @Test
+  public void shouldContinueJobInCaseOfError() {
+    // given
+    final RSAPublicKey validKey = Mockito.mock(RSAPublicKey.class);
+    final String kidForValidKey = "kidForValidKey";
+
+    KeySource keySource =
+        new KeySource() {
+          @Override
+          public List<LoadedPublicKey> loadKeysFromSource() throws KeyLoadFailedException {
+            if (!elseCondition) {
+              return singletonList(new LoadedPublicKey("kidForValidKey", validKey, this));
+            } else {
+              throw new KeyLoadFailedException("Some error occurred during reload");
+            }
+          }
+        };
+
+    keyLoader.addKeySource(keySource);
+    keyLoaderScheduler.start(0, 2, TimeUnit.SECONDS);
+    await().untilAsserted(() -> assertThat(keyLoader.getKey(kidForValidKey)).isSameAs(validKey));
+
+    // when -> simulate error-occurrence:
+    elseCondition = true;
```

Comment:
I think, we have to verify that key loading is scheduled a second time after changing the behavior in this case.